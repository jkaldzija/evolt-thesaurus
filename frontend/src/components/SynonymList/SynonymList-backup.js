import React, {Fragment} from 'react';
import {Typography} from "@material-ui/core";
import type {Word} from "../../api/types";

type SynonymListProps = {
    word: Word
};

const renderSynonyms = (word: Word) => (
    <Fragment>
        <Typography variant="h6" className="synonym-list-title">Synonyms for {word.value}</Typography>
        {word.synonyms.map(synonym => (
            <Typography key={synonym} variant="body1">{synonym}</Typography>
        ))}
    </Fragment>
);

const renderEmpty = (word: Word) => (
    <div className="synonym-empty-container">
        <Typography variant="h5">No synonyms found for {word.value}</Typography>
    </div>
);

const SynonymList = ({word}: SynonymListProps) => {
    if (word.synonyms) {
        return renderSynonyms(word)
    } else {
        return renderEmpty(word);
    }
};

export default SynonymList;
