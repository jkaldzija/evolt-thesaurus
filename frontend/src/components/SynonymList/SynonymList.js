import React, {Fragment} from 'react';
import classnames from "classnames";
import style from "./synonymList.module.css";
import {Link} from "react-router-dom";
import ThemeButton from "../Form/ThemeButton/ThemeButton";
import AddWordDialog from "../AddWord/AddWordDialog";
import {getSynonymRoute} from "../Utility/routing";

type SynonymListProps = {
    synonymList: (Array),
    reloadData: () => void
}


class SynonymList extends React.Component<SynonymListProps> {
    constructor(props) {
        super(props);

        this.state = {
            addSynonymDialogOpen: false
        }
    }

    toggleDialog = (key) => () => {
        const fullKey = `${key}DialogOpen`;
        this.setState({ [fullKey]: !this.state[fullKey] });
    };


    renderSynonymList(){
        const {synonymList} = this.props;
        if(!synonymList || synonymList.length == 0){
            return <div className="d-flex align-items-center mr-3">No synonyms found.</div>
        }
        return <Fragment>
            {synonymList.map((synonym, index) => {
                return <Link key={`${synonym}-${index}`} className={style.link} to={getSynonymRoute(synonym)}>{synonym}</Link>
            })}
        </Fragment>
    }

    render() {
        return <div>
            <div className={classnames("d-flex flex-wrap", style.wrapper)}>
                {this.renderSynonymList()}
                <ThemeButton onClick={this.toggleDialog("addSynonym")} className={style["link-button"]}>
                     Add Synonym
                </ThemeButton>
            </div>
            <AddWordDialog
                word={this.props.currentWord}
                reloadData={this.props.reloadData}
                isOpen={this.state.addSynonymDialogOpen}
                title={"Add synonym"}
                toggle={this.toggleDialog("addSynonym")}
            />
        </div>
    }
}

// Set default props
SynonymList.defaultProps = {
    synonymList:[],
    reloadData: () => {}
};



export default SynonymList;
