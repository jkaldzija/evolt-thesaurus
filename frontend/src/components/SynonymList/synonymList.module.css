.synonym-list-title {
    margin-bottom: 12px !important;
}

.synonym-empty-container {
    display: flex;
    align-items: center;
    justify-content: center;
}

.wrapper{
    margin-bottom: 15px;
}

.link{
    background-color: #88cbe4;
    padding: 5px 10px;
    border-radius: 10px;
    color: white;
    margin-right: 10px;
    font-family: "Museo Sans Rounded 500";
}



.link:hover{
    color: white;
    text-decoration: none;
}

.link-button{
    padding: 5px 10px;
}
