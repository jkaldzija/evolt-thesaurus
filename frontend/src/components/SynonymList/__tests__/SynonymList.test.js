import React from 'react';
import {shallow} from "enzyme";
import SynonymList from "../SynonymList";

describe('Synonym List', function () {
    it('should render correctly', function () {
        const word = {
            value: "word",
            synonyms: [
                "syn1",
                "syn2",
                "syn3"
            ]
        };

        const component = shallow(
            <SynonymList word={word}/>
        );

        expect(component).toMatchSnapshot();
    });
    it('should render correctly for empty', function () {
        const word = {
            value: "word",
            synonyms: null
        };

        const component = shallow(
            <SynonymList word={word}/>
        );

        expect(component).toMatchSnapshot();
    });
});
