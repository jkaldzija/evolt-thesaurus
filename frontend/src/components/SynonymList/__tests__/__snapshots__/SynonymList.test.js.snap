// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Synonym List should render correctly 1`] = `
<div>
  <div
    className="d-flex flex-wrap wrapper"
  >
    <Link
      className="link"
      key="syn1-0"
      replace={false}
      to="/?q=syn1"
    >
      syn1
    </Link>
    <Link
      className="link"
      key="syn2-1"
      replace={false}
      to="/?q=syn2"
    >
      syn2
    </Link>
    <Link
      className="link"
      key="syn3-2"
      replace={false}
      to="/?q=syn3"
    >
      syn3
    </Link>
    <ThemeButton
      className="link-button"
      onClick={[Function]}
    >
      Add Synonym
    </ThemeButton>
  </div>
  <AddWordDialog
    isOpen={false}
    open={false}
    reloadData={[Function]}
    title="Add synonym"
    toggle={[Function]}
    word={
      Object {
        "value": "",
      }
    }
  />
</div>
`;

exports[`Synonym List should render correctly for empty 1`] = `
<div>
  <div
    className="d-flex flex-wrap wrapper"
  >
    <div
      className="d-flex align-items-center mr-3"
    >
      No synonyms found.
    </div>
    <ThemeButton
      className="link-button"
      onClick={[Function]}
    >
      Add Synonym
    </ThemeButton>
  </div>
  <AddWordDialog
    isOpen={false}
    open={false}
    reloadData={[Function]}
    title="Add synonym"
    toggle={[Function]}
    word={
      Object {
        "value": "",
      }
    }
  />
</div>
`;
